sonic-stormond/scripts/stormond

```diff
@@ -0,0 +1,380 @@
+               self.log_info("{} not present.".format(FSIO_RW_JSON_FILE))
+               return
+
+           with open(FSIO_RW_JSON_FILE, 'r') as f:
+               self.fsio_rw_json = json.load(f)
+               self.fsio_json_file_loaded = True
+
+           for storage_device in self.storage.devices:
+               for field in self.statedb_json_sync_fields:
+
+                   value = self.fsio_rw_json[storage_device][field]
+
+                   self.fsio_rw_json[storage_device][field] = "0" if value == "null" else value
+
+        except Exception as e:
+            self.log_error("JSON file could not be loaded: {}".format(str(e)))
+
+        return
+
+
+    # Sync the total and latest procfs reads and writes from STATE_DB to JSON file on disk
+    def _sync_fsio_rw_json(self):
+
+        self.log_info("Syncing total and latest procfs reads and writes from STATE_DB to JSON file")
+
+        json_file_dict = {disk:{} for disk in self.storage.devices}
+        try:
+            for device in self.storage.devices:
+                for field in self.statedb_json_sync_fields:
+                    json_file_dict[device][field] = self.state_db.hget('STORAGE_INFO|{}'.format(device), field)
+            
+            self.fsio_sync_time = time.time()
+            json_file_dict["successful_sync_time"] = str(self.fsio_sync_time)
+
+            with open(FSIO_RW_JSON_FILE, 'w+') as f:
+                json.dump(json_file_dict, f)
+                self.state_db.hset("{}|{}".format(STORAGE_DEVICE_TABLE,FSSTATS_SYNC_TIME_KEY), "successful_sync_time", str(self.fsio_sync_time))
+
+        except Exception as ex:
+            self.log_error("Unable to sync state_db to disk: {}".format(str(ex)))
+
+    # Run a sanity check on the state_db. If successful, get total, latest 
+    # FSIO reads and writes for each storage device from STATE_DB
+    def _load_fsio_rw_statedb(self):
+
+        # Sanity Check:
+        
+        # If the number of STORAGE_INFO|* keys does not equal the 
+        # number of storage disks on the device + FSSTATS_SYNC field,
+        # there has been a corruption to the database. In this case we 
+        # pivot to the JSON file being the Source of Truth.
+        try:
+            if (len(self.state_db.keys("STORAGE_INFO|*")) != (len(self.storage.devices) + 1)):
+                return
+
+            # For each storage device on the switch,
+            for storage_device in self.storage.devices:
+
+                # Get the total and latest procfs reads and writes from STATE_DB
+                for field in self.statedb_json_sync_fields:
+                    value = self.state_db.hget('STORAGE_INFO|{}'.format(storage_device), field)
+                    self.fsio_rw_statedb[storage_device][field] = "0" if value is None else value
+
+                    if value is None: self.log_warning("{}:{} value = None in StateDB".format(storage_device, field))
+            
+            self.statedb_storage_info_loaded = True
```

Comment:
@assrinivasan what if the value is None, in that case we should fall back to .json on the disk

Reply:
Fixed this in latest. Also added a `None` check in the `_load_fsio_rw_json` function for None values. In this scenario, Both StateDB and JSON file have junk values, so it will be considered an init case.